### ISSUE TITLE
pylance consuming 8 GB of memory, crashing

### ISSUE THREAD

**Issue by akrherz** [microsoft/pylance-release#5739]
<h3> Environment data </h3>

-   Language Server version: 2024.4.100
-   OS and version: linux x64
-   Python version: 3.11.8 Anaconda conda-forge 
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

I don't really have any code snippet, it crashes attempting to analyze my workspace/environment

<h3> Repro Steps </h3>

1. launch vscode
2. watch pylance memory usage grow and grow.  Many times reaching 8 GB causing it to crash, presently at 6.4 GB as I type this report.

<h3> Expected behavior </h3>

Like the good ole days, I could just happily run vscode/pylance without specifying a node executable and it would just happily run :)

<h3> Actual behavior </h3>

pylance/node exhausts memory

<h3> Logs </h3>

[plancelogs.txt](https://github.com/microsoft/pylance-release/files/14906094/plancelogs.txt)

This started to become a problem over the past few weeks, so thus why I tried an insider build.

**Comment by akrherz** [comment]
FWIW, I tried stepping back through older pylance extensions back into 2023 even and could not stop the memory growth. Must be something within my heavy environment, but what, shrug yet...

**Comment by akrherz** [comment]
So here's what I have done now, hopefully helpful for others.

1. Removed all repos from my workspace, restarted vscode and watched the node pylance process memory usage, stable at ~600 MB, :heavy_check_mark: 
2. Sequentially added my repos, monitoring the memory usage with each add to ensure the memory usage settles, :heavy_check_mark: 
3. Found a problematic repo with which memory usage grows and grows.  This repo has a two subfolders with non-git tracked testing data (json files and sym links).  Within this repo's `pyproject.toml`, I added the following for those two folders
```toml
[tool.pyright]
autoSearchPaths = false
exclude = ["data/", "prj2wepp/"]
ignore = ["data/", "prj2wepp/"]
```
4. Memory usage currently sitting at 1.4 GB and stable.  :heavy_check_mark: 

Will see if I have jinxed myself and can start using vscode again reliably :)
